src/vtktools.jl

```diff
@@ -42,45 +44,190 @@ end
+  end
+
+  # # Prepare VTK points and cells for celldata file
+  # @timeit "prepare VTK cells (cell data)" begin
+  #   vtk_celldata_points, vtk_celldata_cells = calc_vtk_points_cells(Val(ndims_), coordinates,
+  #                                                                   levels, center_level_0,
+  #                                                                   length_level_0, 1)
+  # end
+
+  # Determine output file names
+  base, _ = splitext(splitdir(filename)[2])
+  vtk_filename = joinpath(output_directory, base)
+  vtk_celldata_filename = vtk_filename * "_celldata"
+
+  # Open VTK files
+  verbose && println("| Building VTK grid...")
+  if is_datafile
+    @timeit "build VTK grid (node data)" vtk_nodedata = vtk_grid(vtk_filename, vtk_points,
+                                                                 vtk_cells)
+  else
+    vtk_nodedata = nothing
+  end
+  # @timeit "build VTK grid (cell data)" vtk_celldata = vtk_grid(vtk_celldata_filename,
+  #                                                               vtk_celldata_points,
+  #                                                               vtk_celldata_cells)
+
+  vtk_celldata = nothing
 
   return vtk_nodedata, vtk_celldata
 end
 
 
+function calc_node_coordinates(mesh::Trixi.CurvedMesh, n_visnodes)
+  # Extract number of spatial dimensions
+  ndims_ = ndims(mesh)
+  n_elements = prod(size(mesh))
+
+  nodes = range(-1, 1, length=n_visnodes)
+  f(args...; kwargs...) = Base.invokelatest(mesh.mapping, args...; kwargs...)
+
+  node_coordinates = Array{Float64, ndims_+2}(undef, ndims_, ntuple(_ -> n_visnodes, ndims_)..., n_elements)
+
+  return calc_node_coordinates!(node_coordinates, f, nodes, mesh)
+end
+
+
+function calc_node_coordinates(mesh::Trixi.UnstructuredQuadMesh, n_visnodes)
+  # Extract number of spatial dimensions
+  ndims_ = ndims(mesh)
+  n_elements = length(mesh)
+
+  # get the uniform nodes
+  nodes = range(-1, 1, length=n_visnodes)
+
+  # intialize the container for the node coordinates
+  node_coordinates = Array{Float64, ndims_+2}(undef, ndims_, ntuple(_ -> n_visnodes, ndims_)..., n_elements)
+
+  # work container for the corners of elements
+  four_corners = zeros(eltype(mesh.corners), 4, 2)
+
+  # loop through all elements and call the correct node coordinate constructor based on curved
+  for element = 1:n_elements
+    if mesh.element_is_curved[element]
+      Trixi.calc_node_coordinates!(node_coordinates, element, nodes, view(mesh.surface_curves, :, element))
+    else # straight sided element
+      for i in 1:4, j in 1:2
+        # pull the (x,y) values of these corners out of the global corners array
+        four_corners[i, j] = mesh.corners[j, mesh.element_node_ids[i, element]]
+      end
+      Trixi.calc_node_coordinates!(node_coordinates, element, nodes, four_corners)
+    end
+  end
+
+  return node_coordinates
+end
+
+
+function calc_node_coordinates!(node_coordinates::AbstractArray{<:Any, 4}, f, nodes, mesh)
```

Comment:
AFAICT, this routine is only useful for the structured mesh and applies for 2D. Thus, I am wondering if one could mention this in a comment to this function (and for the next `calc_node_coordinates!`)